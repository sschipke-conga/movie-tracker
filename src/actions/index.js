export const setMovies = (movies) => ({
  type: 'SET_MOVIES',
  movies,
});

export const setUpcomingMovies = (upcomingMovies) => ({
  type: 'SET_UPCOMING_MOVIES',
  upcomingMovies,
});

export const setUser = (user) => ({
  type: 'SET_USER',
  user,
});

export const setFavorites = (favorites) => ({
  type: 'SET_FAVORITES',
  favorites,
});
