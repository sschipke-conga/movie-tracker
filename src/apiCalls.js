export const getMovies = async () => {
  let res = await fetch('https://api.themoviedb.org/3/discover/movie?sort_by=popularity.desc&api_key=9e1950f701d0af712babdede31811e9e')
  let data = await res.json()
  return data.results
}