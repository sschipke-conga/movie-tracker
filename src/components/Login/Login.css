.screen-cover {
  position:absolute;
  background:rgba(0, 0, 0, 0.8);
  width: 100%;
  height:100vh;

}

.login {
  display: flex;
  justify-content: center;
}


.login__form {
  background: url(../../images/form-background.jpg);
  background-repeat: no-repeat;
  background-size: cover;
  background-position: center;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 100%;
}

input {
  font-size: 20px;
  height: 35px;
  line-height: 30px;
  padding:0 8px;
  width:250px;
  border-radius:5px;
  margin:20px 0;
}

input::placeholder {
  color:white;
  font-size: 15px;
}

.login__btn {
  font-size: 1em;
  height:30px;
  margin-bottom: 20px;
  border-radius: 5px;
  width: 250px;
}

.login__btn:hover {
  cursor: pointer;
  background: red;
  color: white;
}

.error_login {
  color: red;
}

.error {
  border: 2px solid red;
}