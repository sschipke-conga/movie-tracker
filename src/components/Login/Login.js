import React,{ Component } from 'react';


class Login extends Component {
  constructor(){
    super()
    this.state={
      name:'',
      email:'',
      password:''
    }
  }

  render = () => {
    return(
      'Login'
    )
  }


}

export default Login;