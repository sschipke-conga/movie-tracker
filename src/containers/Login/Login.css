.screen-cover {
  position:absolute;
  background:rgba(0, 0, 0, 0.8);
  width: 100%;
  height:100vh;
}

.login {
  display: flex;
  justify-content: center;
  align-items: center;
  background: url(../../images/form-background.jpg);
  background-repeat: no-repeat;
  background-size: cover;
  background-position: center;
}

.login__form {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 800px;
  height:450px;

  border:1px solid rgba(255, 255, 255, 0.5);
  background-color:rgba(0,0,0,0.8)
}

input {
  font-size: 20px;
  height: 35px;
  line-height: 30px;
  padding:0 8px;
  width:250px;
  border-radius:5px;
  margin:20px 0;
}

input::placeholder {
  color:white;
  font-size: 15px;
}

.login__btn {
  font-size: 1em;
  height:30px;
  margin-bottom: 20px;
  border-radius: 5px;
  width: 250px;
}

.login__btn:hover {
  cursor: pointer;
  background: red;
  color: white;
}

.error_login {
  color: red;
}

.error {
  border: 2px solid red;
}