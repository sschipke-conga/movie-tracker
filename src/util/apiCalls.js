export const getMovies = async () => {
  let res = await fetch('https://api.themoviedb.org/3/movie/now_playing?api_key=9e1950f701d0af712babdede31811e9e&language=en-US')
  let data = await res.json()
  return cleanMovieData(data.results)
}

const cleanMovieData = data => data.map(movie => ({
    poster_path: movie.poster_path,
    title: movie.title,
    release_date: movie.release_date,
    vote_average: movie.vote_average,
    overview: movie.overview,
    movie_id: movie.id
  }))

export const getUpcomingMovies = async () => {
  let res = await fetch('https://api.themoviedb.org/3/movie/upcoming?api_key=9e1950f701d0af712babdede31811e9e&language=en-US')
  let data = await res.json()
  return cleanMovieData(data.results)
}

export const createNewUser = async user => {
  let url = 'http://localhost:3001/api/v1/users'
  let options = {
      method: 'POST',
      body: JSON.stringify(user),
        headers: {
      'Content-Type': 'application/json'
    }
  }
  let res = await fetch(url, options);

  if (!res.ok) {
    if(res.status===500) {
      throw Error('This email has already been used')
    }
  }
  return res.json();
}

export const logInUser = async user => {
  let url = 'http://localhost:3001/api/v1/login';
  let options = {
    method: 'POST',
    body: JSON.stringify(user),
    headers: {
      'Content-Type': 'application/json'
    }
  }
  let res = await fetch(url, options)

  if (!res.ok) {
    if (res.status === 401) {
      throw Error('email or password is incorrect')
    }
  }
  return res.json();
}

export const getUserFavorites = async userID => {
  let url = `http://localhost:3001/api/v1/users/${userID}/moviefavorites`;
  let res = await fetch(url);
  let parsedRes = await res.json();
  return parsedRes.favorites;
}

export const postFavorite = async (userId, movie) => {
  console.log('post', userId, movie)
  let url = `http://localhost:3001/api/v1/users/${userId}/moviefavorites`;
  let options = {
    method: 'POST',
    body: JSON.stringify(movie),
    headers: {
      'Content-Type': 'application/json'
    }
  }
  let res = await fetch(url, options);
  if (!res.ok) {
    throw Error('Could not add favorite movie.')
  }
}


export const deleteFavorite = async (userId, movie_id) => {
  console.log('delete', userId, movie_id)
  let url = `http://localhost:3001/api/v1/users/${userId}/moviefavorites/${movie_id}`
  let options = {
    method: 'DELETE',
    headers: {
      'Content-Type': 'application/json'
    }
  }
  let res = await fetch(url, options);
  if (!res.ok) {
      throw Error('Could not delete favorite.')
  }

}
