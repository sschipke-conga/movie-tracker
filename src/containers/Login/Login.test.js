import React from 'react';
import { Login, mapStateToProps } from './Login';
import { shallow } from 'enzyme';
import { createNewUser, logInUser, getUserFavorites } from '../../util/apiCalls';
import * as actions from '../../actions';
import { create } from 'istanbul-reports';

jest.mock('../../util/apiCalls');

describe('Login', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Login />)
  })
  it('should match the initial snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  });
  describe('handleChange', () => {
    it('should update state if a user types in their name', () => {
      const mockEvent = {target: {name: 'name', value: 'Steve' }};
      expect(wrapper.state('name')).toEqual('');
      wrapper.instance().handleChange(mockEvent);
      expect(wrapper.state('name')).toEqual(mockEvent.target.value)
    })
    it('should update state if a user types in their email', () => {
      const mockEvent = { target: { name: 'email', value: 'steve@g.com' } };
      wrapper.instance().setState({name: 'Steve'});
      expect(wrapper.state('email')).toEqual('');
      expect(wrapper.state('name')).toEqual('Steve')

      wrapper.instance().handleChange(mockEvent);
      expect(wrapper.state('email')).toEqual(mockEvent.target.value)
    expect(wrapper.state('name')).toEqual('Steve')
    })
    it('should update state if a user types in their password', () => {
      const mockEvent = { target: { name: 'password', value: 'password' } };
      wrapper.instance().setState({ name: 'Steve' });
      wrapper.instance().setState({ email: 'steve@g.com' });
      expect(wrapper.state('email')).toEqual('steve@g.com');
      expect(wrapper.state('name')).toEqual('Steve')

      wrapper.instance().handleChange(mockEvent);
      expect(wrapper.state('password')).toEqual(mockEvent.target.value);
      expect(wrapper.state('name')).toEqual('Steve');
      expect(wrapper.state('email')).toEqual('steve@g.com')
    })
    it('should run handleChange when the inputs detect a change', () => {
      const mockNameEvent = { target: { name: 'name', value: 'Robbie' } };
      const mockEmailEvent = { target: { name: 'email', value: 's@g.com' } };
      wrapper.instance().handleChange = jest.fn();
      wrapper.instance().forceUpdate();
      wrapper.find('[name="name"]').simulate('change', mockNameEvent);
      wrapper.find('[name="email"]').simulate('change', mockEmailEvent);
      expect(wrapper.instance().handleChange).toHaveBeenCalledWith(mockNameEvent);
      expect(wrapper.instance().handleChange).toHaveBeenCalledWith(mockEmailEvent);
    }) 
  })
  describe('handleSubmit', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = shallow(<Login />)
    })
    it('should prevent the default action when the form is submitted', () => {
      const mockEvent = {preventDefault: jest.fn()};
      wrapper.find('form').simulate('submit', mockEvent);
      expect(mockEvent.preventDefault).toHaveBeenCalled()
    })
  })
  describe('createUser', () => {
    let mockUser = {
      id: 4,
      name: "Susan",
      email: "susan@gmail.com",
      password: "password"
    }
    let mockFaves = [{
      id: 15,
      movie_id: 578189,
      user_id: 4,
      title: "Black and Blue",
      poster_path: "/fjmMu9fpqMMF17mCyLhNfkagKB0.jpg",
      release_date: "2019-10-25",
      vote_average: "4.6",
      overview: "Exposure follows a rookie Detroit African-American female cop who stumbles upon corrupt officers who are murdering a drug dealer, an incident captured by her body cam. They pursue her through the night in an attempt to destroy the footage, but to make matters worse, they've tipped off a criminal gang that she's responsible for the dealer's death."
    }, {
      id: 22,
      movie_id: 454640,
      user_id: 4,
      title: "The Angry Birds Movie 2",
      poster_path: "/ebe8hJRCwdflNQbUjRrfmqtUiNi.jpg",
      release_date: "2019-08-14",
      vote_average: "6.4",
      overview: "Red, Chuck, Bomb and the rest of their feathered friends are surprised when a green pig suggests that they put aside their differences and unite to fight a common threat. Aggressive birds from an island covered in ice are planning to use an elaborate weapon to destroy the fowl and swine."
    }];
    it('should call createNewUser when called', () => {
      let mockNewUser = {
        name: "Susan",
        email: "susan@gmail.com",
        password: "password"
      }
      createNewUser.mockImplementation(() => {
        return Promise.resolve(mockUser)
      }) 
      wrapper.instance().setState({name:'Susan', email: 'susan@gmail.com', password: 'password' });
      wrapper.instance().createUser();
      expect(createNewUser).toHaveBeenCalledWith(mockNewUser);
    })
    it.skip('should update its state to be logged in', () => {
      wrapper.instance().setState({ name: 'Susan', email: 'susan@gmail.com', password: 'password' });
      expect(wrapper.state('isLoggedIn')).toEqual(false);
      wrapper.instance().createUser()
      expect(wrapper.state('isLoggedIn')).toEqual(true);
    });
    it.skip('should set an error message to state if someting goes wrong' , () => {
      createNewUser.mockImplementation(() => {
        return Error('Woops')
      });
      wrapper.instance().createUser();
      // Assertion
    })
  })
  describe('logIn', () => {
    let mockUser = {
      email: 'st@g.com',
      password: 1234
    }
    it('should call logInUser', async () => {
      logInUser.mockImplementation(() => {
        return Promise.resolve(mockUser)
      })
    wrapper.instance().setState({ email: mockUser.email, password:mockUser.password });
    await wrapper.instance().logIn();
    expect(logInUser).toHaveBeenCalledWith(mockUser)
    expect(wrapper.state('isLoggedIn')).toEqual(true)
    })
    it.skip('should set the state if an error occurs', () => {
      logInUser.mockImplementation(() => Promise.reject({message: 'woops'}));
      wrapper.instance().logIn();
      expect(wrapper.state('logInError')).toEqual(true);
    })
  })
  describe('alt snapShot', () => {
    it('should match the snapshot if a user is logged in' ,() => {
      let wrapper = shallow(<Login />);
      wrapper.instance().setState({ isLoggedIn: true });
      expect(wrapper).toMatchSnapshot()
    })
  })
})
