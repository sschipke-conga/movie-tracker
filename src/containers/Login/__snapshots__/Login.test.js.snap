// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login alt snapShot should match the snapshot if a user is logged in 1`] = `
<Redirect
  to="/"
/>
`;

exports[`Login should match the initial snapshot 1`] = `
<div
  className="login screen-cover"
>
  <form
    className="login__form"
    onSubmit={[Function]}
  >
    <input
      maxLength="15"
      name="name"
      onChange={[Function]}
      placeholder="Name"
      required={true}
      value=""
    />
    <input
      className=""
      name="email"
      onChange={[Function]}
      placeholder="Email"
      required={true}
      type="email"
      value=""
    />
    <input
      className=""
      maxLength="20"
      name="password"
      onChange={[Function]}
      placeholder="Password"
      required={true}
      type="password"
      value=""
    />
    <button
      className="login__btn"
      id="create-new"
      type="submit"
    >
      Create an account
    </button>
    <button
      className="login__btn"
      id="login"
      onClick={[Function]}
      type="button"
    >
      Login
    </button>
  </form>
</div>
`;
